app/src/main/java/com/example/androidtraining/ResponseProcessor.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.androidtraining
```

Comment:
This file is the magic sauce. What this whole PR is about. It is in charge of taking a retrofit response and converting it into an error if there was one.